milli/src/update/index_documents/typed_chunk.rs

```diff
@@ -124,13 +125,15 @@ pub(crate) fn write_typed_chunk_into_index(
             let mut operations: Vec<DocumentOperation> = Default::default();
 
             let mut docids = index.documents_ids(wtxn)?;
-            let primary_key = index.primary_key(wtxn)?.unwrap();
-            let primary_key = index.fields_ids_map(wtxn)?.id(primary_key).unwrap();
             let mut cursor = obkv_documents_iter.into_cursor()?;
-            while let Some((docid, reader)) = cursor.move_on_next()? {
+            while let Some((key, reader)) = cursor.move_on_next()? {
                 let mut writer: KvWriter<_, FieldId> = KvWriter::memory();
                 let reader: KvReader<FieldId> = KvReader::new(reader);
-                let docid = docid.try_into().map(DocumentId::from_be_bytes).unwrap();
+
+                let (document_id_bytes, external_id_bytes) = try_split_array_at(key)
+                    .ok_or(SerializationError::Decoding { db_name: Some(DOCUMENTS) })?;
+                let docid = DocumentId::from_be_bytes(document_id_bytes);
+                let external_id = std::str::from_utf8(external_id_bytes)?;
```

Comment:
Now we get the external id from the grenad file instead of trying to retrieve it in the documents and potentially fail if the primary key is nested.